### PR TITLE
feat(365): Add bulk upload

### PR DESCRIPTION
**Pull request by knguyenrise8**
## Description
- add bulk upload
- conditionally render single flow
- adds bulk api call 
## Screenshots (if applicable)
Nov-07-2024 15-24-25
Nov-07-2024 15-23-32

## Related Issues
[Link any related issues or tasks from your project management system.]

## Checklist
- [ ] The title of this PR is descriptive and concise.
- [ ] My changes follow the style guidelines of this project.
- [ ] I have added or updated test cases to cover my changes.
- [ ] I've let the team know about this PR by linking it in the review channel

### CODE REVIEW

File: frontend/src/components/ExtractUploadFile.tsx [hunk 1]
```diff
@@ -23,23 +35,11 @@ export const ExtractUploadFile: React.FC<ExtractUploadFileProps> = ({
...
-    const storedTemplates = localStorage.getItem("templates");
-    if (storedTemplates) {
-      const parsedTemplates = JSON.parse(storedTemplates);
-      setTemplates(parsedTemplates);
-    } else {
-      loadTemplatesTestData();
-    }
-  };
-
+  const [isUploadComplete, setIsUploadComplete] = useState<boolean>(false);
```
Comment (arinkulshi-skylight) [inline frontend/src/components/ExtractUploadFile.tsx:41]: Might need a better state solution soon like contextAPI. This is ok for now.

File: frontend/src/components/FileInput/file-input.tsx [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    })
+  
+    const defaultDragText = multiple
+      ? 'Drag files here or '
+      : 'Drag file here or '
+    const defaultChooseText = 'choose from folder'
+    const defaultErrorText = 'This is not a valid file type.'
+  
+    const filePreviews = []
+    if (files) {
```
Comment (arinkulshi-skylight) [inline frontend/src/components/FileInput/file-input.tsx:98]:
```
const filePreviews = files?.length
  ? files.map((file) => {
      const imageId = makeSafeForID(file.name)
      const key = `filePreview_${imageId}`
      return (
        <FilePreview
          key={key}
          imageId={imageId}
          file={file}
        />
      )
    })
  : []
```

**Review by arinkulshi-skylight — APPROVED** [review]
LGTM!

File: frontend/src/components/ExtractUploadFile.tsx (hunk 1 above)
Comment (knguyenrise8) [reply frontend/src/components/ExtractUploadFile.tsx:41]: sounds good

**Pull request merged**